### PR TITLE
docs: update contribution guide

### PR DESCRIPTION
**Pull request by patzick**
Updates the contribution guidelines to better reflect our current PR workflow and best practices:

* Clarify that PR title (not individual commits) should follow Conventional Commits format
* Add guidance on PR maintenance responsibilities
* Add warning about stale PRs being closed after 2 weeks
* Add recommendation against rebasing public PRs
* Emphasize that PRs are squashed on merge

The changes make the workflow clearer for contributors and align with our current practices.

### CODE REVIEW

**Comment by patzick** [comment]
related: https://github.com/shopware/shopware/pull/5893

File: resources/guidelines/code/contribution.md [hunk 1]
```diff
@@ -25,15 +25,38 @@ To avoid your pull request getting rejected, you should always check that you pr
...
+
+::: tip
+Once your PR is public, avoid rebasing or force-pushing to the branch. Adding new commits makes it easier for reviewers to track changes and see what was updated in response to feedback. The PR will be automatically squashed when merged.
+:::
+
+::: warning
+Pull requests that become stale (no activity from the author for 2 weeks after a review or request for changes) will be closed. You can always reopen the pull request when you're ready to continue working on it.
+:::
+
+Your PR will normally get the first review within a week. We do regular meetings to screen all new pull requests on GitHub. In this meeting, there is a team of Shopware developers of different specializations who will discuss your changes. Together we decide what will happen next to your pull request. We will set one of the following labels, which indicates the status of the pull request. Here is a list of all possible states:
```
Comment (Isengo1989) [inline resources/guidelines/code/contribution.md:59]:
```suggestion
Your PR will normally get the first review within a week. We do regular meetings to screen all new pull requests on GitHub. In this meeting, a team of Shopware developers with different specializations will discuss your changes. Together, we decide what will happen next to your pull request. We will set one of the following labels, which indicates the pull request's status. Here is a list of all possible states:
```

**Review by Isengo1989 — CHANGES_REQUESTED** [review]
Thx for the PR @patzick :tada: 

Some small textual adjustments :writing_hand:

File: resources/guidelines/code/contribution.md [hunk 2]
```diff
@@ -25,15 +25,39 @@ To avoid your pull request getting rejected, you should always check that you pr
...
+* Responding to review comments in a timely manner
+* Updating the code according to review feedback
+* Keeping the PR up to date with the target branch if conflicts arise
+
+::: tip
+Once your PR is public, avoid rebasing or force-pushing to the branch. Adding new commits makes it easier for reviewers to track changes and see what was updated in response to feedback. The PR will be automatically squashed when merged.
+:::
+
+::: warning
+Pull requests that become stale (no activity from the author for 2 weeks after a review or request for changes) will be closed. You can always reopen the pull request when you're ready to continue working on it.
```
Comment (sushmangupta) [inline resources/guidelines/code/contribution.md:57]: @Isengo1989 , is it usually 2 weeks ?

Comment (Isengo1989) [reply resources/guidelines/code/contribution.md:57]: @sushmangupta its for the PRs in shopware/shopware not here I suppose

**Review by Isengo1989 — APPROVED** [review]

**Pull request merged**